src/PriceCalculatorKata/Models/Mess.cs

```diff
@@ -0,0 +1,135 @@
+            }
+        }
+        public DiscountType ProductDiscountType { get; set; } = DiscountType.Additive;
+        private double _upcDiscountPercentage = 0.0;
+        public DiscountPrecedence UPCDiscountPrecedence { get; set; } = DiscountPrecedence.AfterTax;
+        private double _universalDiscountPercentage = 0.0;
+        public double UniversalDiscountPercentage
+        {
+            get { return _universalDiscountPercentage; }
+            set
+            {
+                CheckPercentageValidation(value);
+                _universalDiscountPercentage = value / 100.0;
+            }
+        }
+        public DiscountPrecedence UniversalDiscountPrecedence { get; set; } = DiscountPrecedence.AfterTax;
+        public void CheckPercentageValidation(double value)
+        {
+            if (value < 0 && value > 100)
+                throw new Exception("Percentage is not valid");
+        }
+        private Dictionary<String, double> _additionalCosts = new Dictionary<string, double>();
+        private double Cap = double.MaxValue;
+        public void ReportProductPrice()
+        {
+            double universalDiscountAmount = GetUniversalDiscountAmount();
+            double UPCDiscountAmount = GetUPCDiscountAmount();
+            double totalDiscountAmount = Math.Clamp(universalDiscountAmount + UPCDiscountAmount, 0, Cap);
+            double taxAmount = GetTaxAmount(Price);
+            string regionCurrencySymbol = RegionInfo.CurrencySymbol;
+            string regionISOCurrencySymbol = RegionInfo.ISOCurrencySymbol;
+            Console.WriteLine($"Cost = {regionCurrencySymbol}{Math.Round(Price, 2)} {regionISOCurrencySymbol}");
+            Console.WriteLine($"Tax = {regionCurrencySymbol}{Math.Round(taxAmount, 2)} {regionISOCurrencySymbol}");
+            if (Cap > 0) Console.WriteLine($"Discounts = {regionCurrencySymbol}{Math.Round(totalDiscountAmount, 2)} {regionISOCurrencySymbol}");
+            double additionalCosts = GetAdditionalCosts();
+            double total = Math.Round(Price + taxAmount - totalDiscountAmount + additionalCosts, 2);
+            Console.WriteLine($"Total = {regionCurrencySymbol}{total} {regionISOCurrencySymbol}");
+        }
+        private double GetTaxAmount(double price)
+        {
+            if (UPCDiscountPrecedence == DiscountPrecedence.BeforeTax)
+                price -= GetUPCDiscountAmount();
+            if (UniversalDiscountPrecedence == DiscountPrecedence.BeforeTax)
+                price -= GetUniversalDiscountAmount();
+            double taxAmount = Math.Round(price * TaxPercentage, 4);
+            return taxAmount;
+        }
+        private double GetUniversalDiscountAmount()
+        {
+            double univarsalDiscountAmount = Price * UniversalDiscountPercentage;
+            return Math.Round(univarsalDiscountAmount, 4);
+        }
+        private double GetUPCDiscountAmount()
+        {
+            switch (UPC)
+            {
+                case 12345:
```

Comment:
lets try to resolve this and make it more generic, and not hard code a specific UPC. Try to define a list of UPCs, where we can save the discount for each UPC maybe?